internal/integration/csot_prose_test.go

```diff
@@ -83,6 +84,126 @@ func TestCSOTProse(t *testing.T) {
 		assert.Equal(mt, started[1].CommandName,
 			"insert", "expected a second insert event, got %v", started[1].CommandName)
 	})
+
+	mt.RunOpts("6. gridfs - upload", mtest.NewOptions().MinServerVersion("4.4"), func(mt *mtest.T) {
+		// Drop and re-create the db.fs.files and db.fs.chunks collections.
+		err := mt.Client.Database("db").Collection("fs.files").Drop(context.Background())
+		assert.NoError(t, err, "failed to drop files")
+
+		err = mt.Client.Database("db").Collection("fs.chunks").Drop(context.Background())
+		assert.NoError(t, err, "failed to drop chunks")
+
+		// Set a blocking "insert" fail point.
+		mt.SetFailPoint(mtest.FailPoint{
+			ConfigureFailPoint: "failCommand",
+			Mode: mtest.FailPointMode{
+				Times: 1,
+			},
+			Data: mtest.FailPointData{
+				FailCommands:    []string{"insert"},
+				BlockConnection: true,
+				BlockTimeMS:     15,
+			},
+		})
+
+		// Create a new MongoClient with timeoutMS=10.
+		cliOptions := options.Client().SetTimeout(10).ApplyURI(mtest.ClusterURI())
+		integtest.AddTestServerAPIVersion(cliOptions)
+
+		client, err := mongo.Connect(cliOptions)
+		assert.NoError(t, err, "failed to connect to server")
+
+		// Create a GridFS bucket that wraps the db database.
+		bucket := client.Database("db").GridFSBucket()
+
+		// Note that UploadFromStream accounts for the following steps:
+		// - Call bucket.open_upload_stream()
+		// - Using uploadStream, upload a single 0x12 byte
+		// - Call uploadStream.close() to flush the stream and insert chunks
+		_, err = bucket.UploadFromStream(context.Background(), "filename", bytes.NewReader([]byte{0x12}))
+		assert.ErrorIs(t, err, context.DeadlineExceeded)
+	})
+
+	const test61 = "6.1 gridfs - upload and download with non-expiring client-level timeout"
+	mt.RunOpts(test61, mtest.NewOptions().MinServerVersion("4.4"), func(mt *mtest.T) {
+		// Drop and re-create the db.fs.files and db.fs.chunks collections.
+		err := mt.Client.Database("db").Collection("fs.files").Drop(context.Background())
+		assert.NoError(t, err, "failed to drop files")
+
+		err = mt.Client.Database("db").Collection("fs.chunks").Drop(context.Background())
+		assert.NoError(t, err, "failed to drop chunks")
+
+		// Create a new MongoClient with timeoutMS=10.
+		cliOptions := options.Client().SetTimeout(500 * time.Millisecond).ApplyURI(mtest.ClusterURI())
```

Comment:
Can we use:
```go
.SetTimeout(10 * time.Millisecond)
```
?
Otherwise, it's better to update the comment to match the code.